### DISCUSSION TITLE
v1.8 - Goals and Priorities

### DISCUSSION THREAD

**Discussion by mikekazakov** [mikekazakov/nimble-commander#481] [Development]
Hi,

Here's the current outlook of what I'm aiming to work on for the next version of Nimble Commander:

- [X] Allow additional hotkeys per action.
- [x] Introduce a Gallery-like panel presentation mode.
- [X] Migrate localization files to String Catalogs.
- [x] Expand the User Guide further.
- Bug fixes and other incremental improvements:
  - No specific criteria, as much as possible, though limited in scope.  
    Done so far:
  - #482
  - #485
  - #503
  - #510
  - #513
  - #514
  - #516
  - #550
  - #551
  - #554
  - #562
  - #567
  - #574
  - #584
  - #585
  - #598
  - #600
  - #583
  - #605
  - #614
  - #625

**Comment by evmin** [comment]
Looking forward to it.
Would it be also possible to consider export of the configuration into a file?
Not a reference to settings location but a proper backup.

**Reply by mikekazakov** [reply]
That's not something I'm personally aiming at, but if somebody wants to scratch this particular itch - they are welcome to contribute!

**Reply by yougotwill** [reply]
If this is for your dotfiles, you can use these terminal aliases I made. 🙏

```sh
# nimble commander
alias nc="open -a 'Nimble Commander'"
alias nc-import="cp ~/.config/nimble-commander/Config.json ~/Library/Application\ Support/Nimble\ Commander/Config/Config.json && echo 'nimble commander config imported from ~/.config/nimble-comander/Config.json'"
alias nc-export="cp ~/Library/Application\ Support/Nimble\ Commander/Config/Config.json ~/.config/nimble-commander/Config.json && echo 'nimble commander config exported from ~/Library/Application Support/Nimble Commander/Config/Config.json'"
```

**Comment by mikekazakov** [comment]
Nimble Commander now supports multiple hotkeys (aka shortcuts) per action!
The UI allows setting up to four hotkeys, while only the first two are shown by default:
<img width="842" alt="Screenshot 2024-12-28 at 21 03 41" src="https://github.com/user-attachments/assets/c466ed55-e981-4d75-970b-76411bf2a24b" />

**Comment by yougotwill** [comment]
>  Introduce a Gallery-like panel presentation mode.

Very exciting! Not sure if you have heard of FlowVision but it could be a good source of inspiration so I thought that I would share https://github.com/netdcy/FlowVision

**Reply by mikekazakov** [reply]
This looks nice! For NC I'm thinking towards a horizontal flow, akin to what Finder has been doing for years.

**Reply by yougotwill** [reply]
Ah okay that makes sense if you want match what Finder does. 👍


In that case would you be open to adding support for larger or custom icon sizes in layouts please? 🙏 This way I could fake a media grid layout with larger preview thumbnails. I'm referring to this setting specifically:

![Screenshot 2025-01-07 09 16 46 nZQhmB](https://github.com/user-attachments/assets/2d6b981c-7983-472c-81c2-be8ab6f6976b)


I will make a separate feature request if it's something you think that you will work on in the future?

**Reply by mikekazakov** [reply]
While technically this should be possible, I don't think this could be a good way to emulate a grid layout - to many assumptions will get in the way. If enough users do want a grid layout to be available in NC - well, they can actually implement it - the design of the frontend is not limited to having only two presentation modes.
